src/metabase/api/tiles.clj

```diff
@@ -109,40 +112,59 @@
         (u/ignore-exceptions
           (.close output-stream))))))
 
+(defn- native->source-query
+  "Adjust native queries to be an mbql from a source query so we can add the filter clause."
+  [query]
+  (if (contains? query :native)
+    (let [native (set/rename-keys (:native query) {:query :native})]
+      {:database (:database query)
+       :type     :query
+       :query    {:source-query native}})
+    query))
 
 
 ;;; ---------------------------------------------------- ENDPOINT ----------------------------------------------------
 
+(defn- int-or-string
+  "Parse a string into an integer if it can be otherwise return the string. Intended to determine whether something is a
+  field id or a field name."
+  [x]
+  (try (Integer/parseInt x)
```

Comment:
I don't really love this, but I suppose it's cheaper than checking the string with a regex or something first.

Reply:
the assumption is that it is either an integer or a column name. We could switch it to `(if (re-matches #"\d*" x))` but i suspect that is basically what parseInt is doing. Happy to change if you have a strong opinion.